src/io/obf.rs

```diff
@@ -352,38 +314,56 @@ pub async fn obfuscate<M, SU, SH, ST, R, P>(
     log_mem("Decomposed RLWE ciphertext into {BaseDecompose(a), BaseDecompose(b)}");
     handles.push(store_and_drop_matrix(b, &dir_path, "b"));
 
-    let final_preimage_target = {
+    // P_att := u_1_L' ⊗ A_att - I_L' ⊗ G_n+1
+    let gadget = M::gadget_matrix(&params, d + 1);
+    let final_preimage_target_att =
+        u_1_l.tensor(&pub_key_att_matrix) - identity_1_plus_packed_input_size.tensor(&gadget);
+    let final_preimage_att = sampler_trapdoor.preimage(
```

Comment:
Can we rename it to `final_preimage_lhs` ? And, similarly, rename `final_preimage_f` to `final_preimage_rhs`. 

I suggest this change because in the paper the actual "whole" preimage is defined as `K_f` which includes both the LHS (using p_att as target) and RHS (using p_f as target)

Reply:
lets match naming of paper to the implementation in this case